src/vehicle_use.cpp

```diff
@@ -1744,6 +1744,136 @@ void vehicle::use_harness( int part, const tripoint &pos )
     }
 }
 
+std::vector<item_count_tuple> vehicle::get_comestible_liquids()
+{
+    std::vector<item_count_tuple> liquids;
+    std::map<item *, uint32_t> liquids_map;
+
+    liquids.clear();
+    liquids_map.clear();
+
+    for( auto &part : parts ) {
+        const int remaining = part.ammo_remaining();
+
+        if( !part.is_tank() || part.base.contents_made_of( SOLID ) || remaining <= 0 ) {
+            continue;
+        }
+
+        item *food_ptr = part.base.get_food();
+
+        if( food_ptr == nullptr || !food_ptr->is_comestible() ) {
+            continue;
+        }
+
+        const cata::value_ptr<islot_comestible> comestible = food_ptr->get_comestible();
+
+        if( comestible == nullptr || "DRINK" != comestible->comesttype ) {
+            continue;
+        }
+
+        const itype_id liquid_id = part.ammo_current();
+
+        if( liquid_id.empty() || liquid_id == "null" || liquid_id == "water" ) {
+            continue;
+        }
+
+        if( liquids_map.find( food_ptr ) == liquids_map.end() ) {
+            liquids_map[food_ptr] = static_cast<uint32_t>( remaining );
+        } else {
+            liquids_map[food_ptr] += static_cast<uint32_t>( remaining );
+        }
+    }
+
+    for( auto &liquid : liquids_map ) {
+        liquids.push_back( item_count_tuple( liquid.first, liquid.second ) );
+    }
+
+    std::sort( liquids.begin(), liquids.end(),
+    []( const item_count_tuple & a, const item_count_tuple & b ) {
+        return std::get<1>( a ) > std::get<1>( b );
+    } );
+
+    return liquids;
+}
+
+item_count_tuple vehicle::select_comestible_liquid( const std::vector<item_count_tuple> &liquids,
+        const std::string &message )
+{
+    uilist selectmenu;
+    int choice = std::numeric_limits<int>::min();
+    int idx = 0;
+
+    if( !liquids.empty() ) {
```

Comment:
```suggestion
    if( liquids.empty() ) { return ...; }
```
And the remaining code does not need to be indented.